include/simdjson/generic/ondemand/array.h

```diff
@@ -43,6 +43,10 @@ class array {
    * safe to continue.
    */
   simdjson_really_inline simdjson_result<size_t> count_elements() & noexcept;
+
+  inline simdjson_result<value> at_pointer(std::string_view json_pointer) noexcept;
+  simdjson_really_inline simdjson_result<value> at(size_t index) noexcept;
+
```

Comment:
If this is a public method then it should be described with a comment above. In this instance, the comment should say that this is a linear-time function, not a constant-time function.

Reply:
And if we are making `at` public, then I think we need to have at least one corresponding test for just this function (to make sure it behaves as we expect). For example, do we think that ...

```
auto v1 = myarray.at(1);
auto v2 = myarray.at(10);
```

would work?

It is almost surely the case that...
```
auto v1 = myarray.at(10);
auto v2 = myarray.at(1);
```
will not work as expected (and that should be documented).

If this gets too complicated, I recommend making the function `protected` so that random users do not try to use it. If we think we should have a nice `at` function, and we are not quite ready to make it public, we can open an issue.

cc @jkeiser

Reply:
I don't think we should make random accesses public in general. It sends the wrong message about the API in general. If you need it public so that other classes can access it, I suggest using the friend syntax (we do this for other stuff).

Reply:
@jkeiser It was now made `protected` but @NicolasJiaxin was nice enough to document it.